code/modules/library/lib_machines.dm

```diff
@@ -406,7 +406,11 @@ var/global/list/datum/cachedbook/cachedbooks // List of our cached book datums
 						else
 							log_game("[usr.name]/[usr.key] has uploaded the book titled [scanner.cache.name], [length(scanner.cache.dat)] signs")
 							alert("Upload Complete. Uploaded title will be unavailable for printing for a short period")
-
+	if(href_list["orderbyid"])
+		var/orderid = input("Enter your order:") as num|null
+		if(orderid)
+			if(isnum(orderid))
+				href_list["targetid"] = orderid
 	if(href_list["targetid"])

```

Comment:
Have you tested this? I'm not 100% sure of the semantics of this in DM... Does it work like C's `switch()`es and fall down to the next one unless returned?